### PR TITLE
Schema config documentation

### PR DESCRIPTION
**Pull request by hkir-dev**
Related with issue #370

New documentation: https://github.com/INCATools/ontology-development-kit/blob/issue-370/docs/project-schema.md

Generates ODK schema config documentation from the 'project-schema.json'

Schema json doesn't contain field descriptions so extracts those descriptions from 'odk.py' using ast.

Generates the initial (plain) documentation using jsonschema2md library. Then makes documentation more readable through
in-lining referred objects and 'allOf' definitions.

### CODE REVIEW

**Comment by cmungall** [comment]
looks good!

Did we already consider linkml for this?

**Comment by matentzn** [comment]
No we are still using your old odk.py for generating schema after the fact. The source of truth are the python data classes. While you could probably generate a linkml model from the json generated from the data classes, it would take a long time to replace the whole pipeline with a purely linkml-based approach. It would be cool of course, especially since I would like to enable other ontology tools like Protege or DROID to refer to this as a standard representation of onto repo config, but.. one urgent plaster on a bloody patch at a time.

File: requirements.txt [hunk 1]
```diff
@@ -35,3 +35,4 @@ tsvalid
 j2cli[yaml]
```
Comment (matentzn) [inline requirements.txt:35]: Rename file back to what it was, as the original name is needed to build ODK!

**Review by matentzn — CHANGES_REQUESTED** [review]
Excellent starting point I think! After fixing the one issue, I will mere this is. What do you think @gouttegd - Given there was _nothing_ at all like this in ODK previously, I think this is a huge step forward!

@hkir-dev can you also just publish the markdown straight in the docs/ directory please?

File: requirements.txt (hunk 1 above)
Comment (gouttegd) [reply requirements.txt:35]: FYI, the reason why this file is not named `requirements.txt` is given in commit 3fdd6bf1ada6c4e3ee9b2c7190e2660a69bd7732.

**Comment by gouttegd** [comment]
@matentzn _Some_ documentation is always better than _no_ documentation. :) Looks good for me, modulo the `requirements.txt.full` issue.

**Comment by matentzn** [comment]
Thanks @gouttegd :)

File: requirements.txt (hunk 1 above)
Comment (hkir-dev) [reply requirements.txt]: Thank you, seems I forgot to restore the file name when development finished.And updated default output location to docs/ directory now.

**Review by gouttegd — APPROVED** [review]

**Review by matentzn — APPROVED** [review]

**Comment by matentzn** [comment]
Thank you @hkir-dev super cool.

**Pull request merged**